app/assets/javascripts/application/components/lesson_plan.js.jsx

```diff
@@ -40,6 +26,31 @@ var LessonPlan = createReactClass({
     });
   },
 
+  reorderLessons: function({oldIndex, newIndex}) {
+    // Optimistically update during AJAX
+    this.setState({
+      lessons: arrayMove(this.state.lessons, oldIndex, newIndex)
+    });
+
+    this.persistLessonPlan({
+      lesson_plan_lessons_attributes: this.state.lessons.map((el, index) => {
```

Comment:
This should go in a callback passed to `setState`:

> setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall.

https://reactjs.org/docs/react-component.html#setstate